apps/emqx_dashboard/src/emqx_dashboard_api.erl

```diff
@@ -211,8 +213,14 @@ user(delete, #{bindings := #{username := Username}}) ->
         true -> {400, #{code => <<"CONNOT_DELETE_ADMIN">>,
                         message => <<"Cannot delete admin">>}};
         false ->
-            _ = emqx_dashboard_admin:remove_user(Username),
-            {200}
+            case emqx_dashboard_admin:lookup_user(Username) of
+                [] ->
+                    Message = list_to_binary(io_lib:format("Username [~p] not exist", [Username])),
+                    {404, 'NOT_FOUND', Message};
```

Comment:
make the API idempotent, return `200` with a `not_found` message.

Reply:
Returning 404 was the result of discussion and is now a common API practice. I agree with you, but maybe i cannot change it.

Reply:
I think we need to discuss it again when I finish a new EIP.

Reply:
ok, let's keep it this way then.